### PR TITLE
Address issues 

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
Address two issues:

1) #40764 : bug in resize of a vector;

2) #40477 : clean-up of the old model class for R-hadrons hadronic interactions, removed old commented lines, removed unused variables, substitute internal Poisson method by Geant4 function, simplified count of the number of nucleons, substitute exit(..) by G4Exception(..).

No change in any WF is expected. 

#### PR validation:
private

### CODE REVIEW

**Comment by civanch** [comment]
type bugfix

**Comment by civanch** [comment]
please test

File: DataFormats/EcalDigi/src/EcalTimeDigi.cc [hunk 1]
```diff
@@ -10,11 +10,9 @@ EcalTimeDigi::EcalTimeDigi(const DetId& id)
     : id_(id), size_(0), sampleOfInterest_(-1), waveform_(WAVEFORMSAMPLES), data_(MAXSAMPLES) {}
 
 void EcalTimeDigi::setSize(unsigned int size) {
+  size_ = size;
   if (size > MAXSAMPLES)
-    size_ = MAXSAMPLES;
-  else
-    size_ = size;
-  data_.resize(size_);
+    data_.resize(size_);
```
Comment (makortel) [inline DataFormats/EcalDigi/src/EcalTimeDigi.cc:15]: I just want to note that according to https://github.com/cms-sw/cmssw/issues/40764#issuecomment-1507230611 the silent cap of `MAXSAMPLES` does not seem to be the cause for https://github.com/cms-sw/cmssw/issues/40764

**Comment by makortel** [comment]
test parameters:
  - release = CMSSW_13_1_ASAN_X
  - workflow =  21034.114

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by civanch** [comment]
please test

it seems there was a real problem in fill access of arrays. This may change results, so we need to runn standard set of tests first.

**Comment by wddgit** [comment]
The unit test failure (TestTriggerResultsFormat) is probably caused by the fact that the base IB release for this PR test invocation precedes the release where this unit test was added. The unit test depends on a new data file added to the new external data area. I think it will pick it up if you rerun the tests again. ASAN with CMSSW_13_1_X_2023-05-01-2300 should have it.

**Comment by civanch** [comment]
It was a mistake to try to fix two issues in one PR. Let us test run but this will be closed and two new will be opened. Relatively simple fix #41487 is already submitted. Another will be proposed for DIGI step.

**Pull request closed without merging**